src/test/java/com/hazelcast/commandline/member/MemberCommandLineTest.java

```diff
@@ -0,0 +1,193 @@
+
+        Files.walk(pathToBeDeleted)
+                .sorted(Comparator.reverseOrder())
+                .map(Path::toFile)
+                .forEach(File::delete);
+    }
+
+    private void killAllRunningHazelcastInstances() throws IOException, InterruptedException {
+        String out = getRunningJavaProcesses();
+        List<String> pids = getHazelcastInstancePids(out);
+        runCommand("/bin/kill -9 " + String.join(" ", pids));
+
+    }
+
+    private List<String> getHazelcastInstancePids(String out) {
+        List<String> pids = new ArrayList<>();
+        for (String line : out.split("\n")) {
+            if (line.contains(HazelcastMember.class.getSimpleName())) {
+                pids.add(line.split(" ")[0]);
+            }
+        }
+        return pids;
+    }
+
+    @Test(timeout = 10000)
+    public void test_start() throws IOException, ClassNotFoundException, InterruptedException {
+        memberCommandLine.start(null, null, null, null,
+                false, null, null);
+        assertTrue(memberCommandLine.getProcessOutput().anyMatch(out ->
+                out.contains(LifecycleEvent.LifecycleState.STARTED.toString())));
+    }
+
+    @Test(timeout = 10000)
+    public void test_start_withConfigFile() throws IOException, ClassNotFoundException, InterruptedException {
+        String groupName = "member-command-line-test";
+        startMemberWithConfigFile();
+        assertTrue(memberCommandLine.getProcessOutput().anyMatch(out ->
+                out.contains(groupName) && out.contains(LifecycleEvent.LifecycleState.STARTED.toString())));
+    }
+
+    @Test(timeout = 10000)
+    public void test_start_withClusterName() throws IOException, ClassNotFoundException, InterruptedException {
+        String groupName = "member-command-line-test";
+        memberCommandLine.start(null, groupName, null, null, false,
+                null, null);
+        assertTrue(memberCommandLine.getProcessOutput().anyMatch(out ->
+                out.contains(groupName) && out.contains(LifecycleEvent.LifecycleState.STARTED.toString())));
+    }
+
+    @Test(timeout = 10000)
+    public void test_start_withPort() throws IOException, ClassNotFoundException, InterruptedException {
+        String port = "9898";
+        memberCommandLine.start(null, null, port, null,
+                false, null, null);
+        assertTrue(memberCommandLine.getProcessOutput().anyMatch(out ->
+                out.contains(port + " is " + LifecycleEvent.LifecycleState.STARTED.toString())));
+    }
+
+    @Test
+    public void test_start_withJVMOptions() throws IOException, ClassNotFoundException, InterruptedException {
+        List<String> javaOptions = new ArrayList<>();
+        String option = "-Xmx128m";
+        javaOptions.add(option);
+        memberCommandLine.start(null, null, null, null,
+                false, null, javaOptions);
+        String output = runCommand("jps -v");
```

Comment:
This is tricky, because you check all HazelcastMember processes, not necessarily the one you started. So your test is fragile and may give false positive.

Reply:
Yes, I know this issue. Do you have any suggestions?